src/codegen/llvm/codegen_llvm.cc

```diff
@@ -100,7 +101,7 @@ void CodeGenLLVM::AddFunctionInternal(const LoweredFunc& f, bool ret_void) {
     Type t = arg.type();
     if (t.is_handle() && f->handle_data_type.count(arg)) {
       arg_type.push_back(
-          LLVMType(f->handle_data_type[arg].type())->getPointerTo());
+          LLVMType(f->handle_data_type[arg].type())->getPointerTo(isTargetAMD ? 1 : 0));
```

Comment:
Add a virtual function GetGlobalAddressSpace to CodeGenLLVM and override that in CodeGenAMDGPU